packages/packagers/webextension/src/WebExtensionPackager.js

```diff
@@ -56,9 +58,16 @@ export default (new Packager({
         .map(relPath);
 
       if (resources.length > 0) {
-        // In the future, maybe use "matches" as well
         war.push({
-          extension_ids: [],
+          matches: contentScript.matches.map(match => {
+            if (/^(((http|ws)s?)|ftp|\*):\/\//.test(match)) {
+              let pathIndex = match.indexOf('/', match.indexOf('://') + 3);
+              // Avoids creating additional errors in invalid match URLs
+              if (pathIndex == -1) pathIndex = match.length;
+              return match.slice(0, pathIndex) + '/*';
+            }
+            return match;
+          }),
```

Comment:
So if I'm reading this right, "https://github.com/fregante/browser-extension-template/*" would be transformed to "https://github.com/*"? What's the reason for that?

We could also consider using a URL parser for this:

```js
  return new URL(match).origin + '/*'
```

Reply:
In web extensions `web_accessible_resources` will error if the path is not a wildcard. We can't use the `new URL` constructor because there's a bunch of special syntax here like `*://something*.com/*` that doesn't get parsed properly by `new URL` but is covered by this logic.

Now in theory we could literally replace all this with `["<all_urls>"]` and it would work the same, but that makes fingerprinting by bad actors much easier.